HLTrigger/special/src/HLTPixelIsolTrackL1TFilter.cc

```diff
@@ -0,0 +1,131 @@
+  candTag_             = iConfig.getParameter<edm::InputTag> ("candTag");
+  hltGTseedlabel_      = iConfig.getParameter<edm::InputTag> ("L1GTSeedLabel");
+  minDeltaPtL1Jet_     = iConfig.getParameter<double> ("MinDeltaPtL1Jet");
+  minpttrack_          = iConfig.getParameter<double> ("MinPtTrack");
+  maxptnearby_         = iConfig.getParameter<double> ("MaxPtNearby");
+  maxetatrack_         = iConfig.getParameter<double> ("MaxEtaTrack");
+  minetatrack_         = iConfig.getParameter<double> ("MinEtaTrack");
+  filterE_             = iConfig.getParameter<bool> ("filterTrackEnergy");
+  minEnergy_           = iConfig.getParameter<double>("MinEnergyTrack");
+  nMaxTrackCandidates_ = iConfig.getParameter<int>("NMaxTrackCandidates");
+  dropMultiL2Event_    = iConfig.getParameter<bool> ("DropMultiL2Event");
+  candToken_ = consumes<reco::IsolatedPixelTrackCandidateCollection>(candTag_);
+  hltGTseedToken_ = consumes<trigger::TriggerFilterObjectWithRefs>(hltGTseedlabel_);
+}
+
+HLTPixelIsolTrackL1TFilter::~HLTPixelIsolTrackL1TFilter(){}
+
+
+void
+HLTPixelIsolTrackL1TFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+  desc.add<edm::InputTag>("candTag",edm::InputTag("isolPixelTrackProd"));
+  desc.add<edm::InputTag>("L1GTSeedLabel",edm::InputTag("hltL1sIsoTrack"));
+  desc.add<double>("MaxPtNearby",2.0);
+  desc.add<double>("MinEnergyTrack",15.0);
+  desc.add<double>("MinPtTrack",20.);
+  desc.add<double>("MaxEtaTrack",1.9);
+  desc.add<double>("MinEtaTrack",0.0);
+  desc.add<double>("MinDeltaPtL1Jet",4.0);
+  desc.add<bool>("filterTrackEnergy",true);
+  desc.add<int>("NMaxTrackCandidates",10);
+  desc.add<bool>("DropMultiL2Event",false);
+  descriptions.add("isolPixelTrackFilter",desc);
+}
+
+bool HLTPixelIsolTrackL1TFilter::hltFilter(edm::Event& iEvent, const edm::EventSetup& iSetup, trigger::TriggerFilterObjectWithRefs & filterproduct) const {
+
+  if (saveTags())
+    filterproduct.addCollectionTag(candTag_);
+  
+  // Ref to Candidate object to be recorded in filter object
+  edm::Ref<reco::IsolatedPixelTrackCandidateCollection> candref;
+
+  // get hold of filtered candidates
+  edm::Handle<reco::IsolatedPixelTrackCandidateCollection> recotrackcands;
+  iEvent.getByToken(candToken_,recotrackcands);
+
+  //Filtering
+
+  //find leading L1 jet:
+  double ptTriggered  = -10;
+
+  edm::Handle<trigger::TriggerFilterObjectWithRefs> l1trigobj;
+  iEvent.getByToken(hltGTseedToken_, l1trigobj);
+
+//  std::vector< edm::Ref<l1t::L1JetParticleCollection> > l1tauobjref;
+//  std::vector< edm::Ref<l1t::L1JetParticleCollection> > l1jetobjref;
+//  std::vector< edm::Ref<l1t::L1JetParticleCollection> > l1forjetobjref;
+  std::vector< edm::Ref<l1t::JetBxCollection> > l1tauobjref;
+  std::vector< edm::Ref<l1t::JetBxCollection> > l1jetobjref;
+  std::vector< edm::Ref<l1t::JetBxCollection> > l1forjetobjref;
+
```

Comment:
As above - there are only Jet and Tau types and collections, each with its own C++type